File: BackEnd/Api/Application/DataTransfer/PhysicalResourceDto.cs
namespace Api.Application.DataTransfer;

using Api.Domain.ValueObjects;
using Api.Domain.Entities;

public class PhysicalResourceDto
{
    public required string Code { get; set; }
    public required string Description { get; set; }
    public required ResourceStatus Status { get; set; }
    public required int SetupTimeInMinutes { get; set; }
    public required List<QualificationDto> Qualifications { get; set; }
    public required OperationalWindow OperationalWindow { get; set; }
}

public class STSCraneDto : PhysicalResourceDto
{
    public required uint ContainersPerHour { get; set; }
    public required uint LiftingCapacity { get; set; }
    public required DockDto ServingDock { get; set; }
}

public class YardCraneDto : PhysicalResourceDto
{
    public required uint ContainersPerHour { get; set; }
    public required uint LiftingCapacity { get; set; }
    public required StorageAreaDto YardSection { get; set; }
}

public class TruckDto : PhysicalResourceDto
{
    public required uint ContainersPerTrip { get; set; }
    public required uint AverageSpeed { get; set; } 
    public required uint MaxLoadCapacity { get; set; }
}

File: FrontEnd/src/model/dto/PhysicalResourceDto.ts
import type { OperationalWindow } from "../OperationalWindow";

export interface PhysicalResourceDto {
    code: string;
    description: string;
    status: number;
    setupTimeInMinutes: number;
    operationalWindow: OperationalWindow;
    qualificationsCodes: string[];
}

export interface STSCraneDto extends PhysicalResourceDto {
    liftingCapacity: number;
    servingDockCode: string;
    containersPerHour: number;
}

export interface YardCraneDto extends PhysicalResourceDto {
    liftingCapacity: number;
    containersPerHour: number;
}

export interface TruckDto extends PhysicalResourceDto {
    maxLoadCapacity: number;
    averageSpeed: number;
    containersPerTrip: number;
}

export interface PhysicalResourceFilter {
    Code?: string;
    Description?: string;
    Status?: 0 | 1 | 2;
    Type?: 0 | 1 | 2;
}